projects/pancakeswap/index.js

```diff
@@ -1,5 +1,236 @@
+  })).output;
+
+  let rewardForPool = rewardTokens.reduce((acc, res) => {
+    if (res.success) {
+      const { target: poolAddress } = res.input;
+      // Here we make the (valid) assumption that no two pools have the same reward
+      acc[poolAddress] = res.output;
+    }
+
+    return acc;
+  }, {});
+
+  // Get reward token balances
+  const rewardBalances = (await sdk.bsc.abi.multiCall({
+    abi: "bep20:balanceOf",
+    block, 
+    calls: Object.entries(rewardForPool)
+      .map(([poolAddress, rewardAddress]) => ({ target: rewardAddress, params: poolAddress }))
+  })).output;
+
+  // Combine addresses with balances
+  return rewardBalances.reduce((acc, res) => {
+    const { target: rewardAddress } = res.input;
+
+    if (res.success && rewardAddress.toLowerCase() !== CAKE.toLowerCase()) {
+      acc[rewardAddress] = res.output;
+    }
+
+    return acc;
+  }, {});
+}
+
+async function getPoolDeposits(block) {
+  const balances = (await sdk.bsc.abi.multiCall({
+    target: CAKE,
+    abi: BALANCE_OF,
+    block,
+    calls: SYRUP_POOLS.map(address => ({ params: address }))
+  })).output;
+
+  return balances.reduce((acc, res) => {
+    if (res.success) {
+      const existingBalance = new BigNumber(acc[CAKE] ?? '0');
+      const additionalBalance = new BigNumber(res.output);
+
+      acc[CAKE] = existingBalance.plus(additionalBalance).toFixed();
+    }
+
+    return acc;
+  }, {});
+}
+
+async function getPoolBalances(block) {
+  const poolRewards = await getPoolRewards(block);
+  const poolDeposits = await getPoolDeposits(block);
+
+  return combineBalances(poolDeposits, poolRewards);
+}
+
+function combineBalances(b1, b2) {
+  const normal1 = Object.entries(b1).reduce((acc, [key, value]) => {
+    acc[key.toLowerCase()] = value;
+    return acc;
+  }, {});
+  const normal2 = Object.entries(b2).reduce((acc, [key, value]) => {
+    acc[key.toLowerCase()] = value;
+    return acc;
+  }, {});
+
+  const addressSet = new Set([...Object.keys(normal1), ...Object.keys(normal2)]);
+  const addresses = Array.from(addressSet);
+
+  return addresses.reduce((acc, address) => {
+    const balance1 = new BigNumber(normal1[address] ?? '0');
+    const balance2 = new BigNumber(normal2[address] ?? '0');
+    acc[address] = balance1.plus(balance2).toFixed();
+
+    return acc;
+  }, {});
+}
+
+async function tvl(_, block) {
+  const pairAddresses = await fetchPairAddresses()
+
+  const poolBalances = await getPoolBalances(block);
+  const reserveBalances = await sdk.bsc.swap.getReservedBalances(pairAddresses);
+
+  const balances = combineBalances(poolBalances, reserveBalances);
+
+  return (await sdk.bsc.util.toSymbols(balances)).output
+}
+
+module.exports = {
+  version: '2', // to distinguish new version from old version
+  name: 'PancakeSwap',
+  token: 'CAKE',
+  category: 'dexes',
+  start: 1600753669, // Sep-22-2020 05:47:49 AM +UTC
+  tvl
+}
+
+// Last fetched Apr-20-2020
+const SYRUP_POOLS = [
```

Comment:
The only place I could find these addresses were in a source code file on GitHub. They are also created by seemingly random user addresses.